lib/open_project/translations/models/git_repository.rb

```diff
@@ -0,0 +1,73 @@
+require_relative '../helpers/run_command'
+
+class GitRepository
+  include RunCommand
+
+  def initialize(uri, path)
+    @uri = uri
+    @path = path
+  end
+
+  def clone_or_pull
+    if File.directory? File.join(@path, '.git')
+      within_repo do
+        run_command "git fetch"
+      end
+    #elsif File.directory? @path
+      # maybe we should:
+      # raise "Cannot checkout #{@uri} to #{@path}. Directory already exists but has no .git folder."
+      # or ask if the directory should be overwritten
+    else
+      run_command "git clone #{@uri} #{@path}"
```

Comment:
The method is named  `clone_or_pull` while you are actually doing `clone_or_fetch`. What I am getting at is the different outcome of those two code paths. You probably want to use `git pull` if the repo exists.

Note however, that a `git pull` can fail if you have local changes. 

I haven't read any further so maybe you are using it differently but it might be a good idea to always start with `git clone`